retention messaging (#422)
adaptyteam/adapty-docs: src/content/docs/version-3.0/retention-messaging.mdx

---
title: "Retention Messaging"

> description: "Show subscribers a custom message inside Apple's cancel-subscription screen — you author it in Adapty, and Adapty delivers it to Apple in real time."

Reviewer: > cancel-subscription screen

мы везде по-разному называем чёт. давай везде Cancel Subscription screen использовать

metadataTitle: "Apple Retention Messaging | Adapty Docs"
---

<CustomDocCardList ids={['retention-messaging-setup', 'retention-messaging-create']} />

> Apple's Retention Messaging API shows a custom message on the iOS Cancel Subscription screen, before a subscriber confirms the cancellation. Adapty allows you to author the message in the dashboard and deliver it to Apple automatically. A subscriber who taps Cancel sees your message the moment they're about to leave.

Reviewer: Сорри, буду придираться к словам, иначе потом это сделает Юра

> Apple's Retention Messaging API shows

не API shows, а фича под названием Retention Messaging

> A subscriber who taps Cancel sees your message the moment they're about to leave.

оч драматично + дублирует первое предложение частично

моё предложение (можешь и по-другому сделать, эт как пример)

```suggestion
Retention Messaging shows subscribers a custom message on the iOS Cancel Subscription screen before they confirm the cancellation. You write the message in the Adapty dashboard, and Adapty delivers it to Apple automatically. This feature uses Apple's Retention Messaging API.
```

## How it works

> Adapty sits between your dashboard and Apple:

Reviewer: Странная метафора, да и по факту — разве дашборд не Адапти?
Можно без вводного предложения или простое типа here's the flow

1. You author a message in Adapty and submit it for Apple's review.
2. Adapty uploads the approved message to Apple's servers.
3. When a subscriber taps Cancel in iOS Settings, Apple calls Adapty in real time to retrieve the message for that product and locale.

> 4. Apple displays the message received from Adapty on in the cancel screen.

Reviewer: ```suggestion
4. Apple displays the message on the cancel screen.
```

## Requirements

> - Retention Messaging is an **iOS exclusive** feature that uses Apple's Retention Messaging API.

Reviewer: Это не совсем требование... Я б просто убрала

- Your app must have access to [Apple's Retention Messaging program](https://developer.apple.com/documentation/retentionmessaging).
- You need an App Store Connect API key in Adapty. Adapty uses it to register your realtime URL and upload messages to Apple.

## Get started

- **[Set up the connection](retention-messaging-setup)**: Connect Apple's Retention Messaging realtime URL to Adapty, test in sandbox, then run in production.

> - **[Author a default message](retention-messaging-create)**: Write a title and body, and optionally add an image. Adapty shows it when a subscriber cancels, and translates it into your other locales.

Reviewer: > Adapty shows it when a subscriber cancels, and translates it into your other locales.

всё-таки Apple shows

Reviewer: порядок ещё странный в последнем предложении

лучше Adapty translates the message into your other locales and delivers it to Apple.
